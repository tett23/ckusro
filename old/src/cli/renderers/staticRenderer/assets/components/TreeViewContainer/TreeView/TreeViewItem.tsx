import React from 'react';
import {
  FileBuffer,
  fileBufferName,
} from '../../../../../../../models/FileBuffer';

export type Props = {
  fileBuffer: FileBuffer;
};

export default function TreeViewItem({ fileBuffer }: Props) {
  return (
    <li>
      <div>{fileBufferName(fileBuffer)}</div>
    </li>
  );
}
