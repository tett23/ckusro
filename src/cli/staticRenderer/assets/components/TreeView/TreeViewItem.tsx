import { join } from 'path';
import React, { ReactNode } from 'react';
import {
  FileBuffer,
  fileBufferName,
  replaceExt,
} from '../../../../../models/FileBuffer';
import { OutputContext } from '../../../../../models/OutputContext';

export type Props = {
  file: FileBuffer;
  context: OutputContext;
  children?: ReactNode;
};

export default function TreeViewItem({ context, file, children }: Props) {
  return (
    <li>
      <div>
        <a href={join(context.path, replaceExt(file))}>
          {fileBufferName(file)}
        </a>
      </div>
      <Children items={children} />
    </li>
  );
}

function Children({ items }: { items: ReactNode }) {
  if (items == null) {
    return null;
  }

  return <ul>{items}</ul>;
}
