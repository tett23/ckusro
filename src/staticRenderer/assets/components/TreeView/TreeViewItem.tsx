import { join } from 'path';
import React, { ReactNode } from 'react';
import { LoaderContext } from '../../../../loader';
import { CkusroFile, replaceExt } from '../../../../models/ckusroFile';

export type Props = {
  file: CkusroFile;
  context: LoaderContext;
  children?: ReactNode;
};

export default function TreeViewItem({ context, file, children }: Props) {
  return (
    <li>
      <div>
        <a href={join(context.path, replaceExt(file))}>{file.name}</a>
      </div>
      <Children items={children} />
    </li>
  );
}

function Children({ items }: { items: ReactNode }) {
  if (items == null) {
    return null;
  }

  return <ul>{items}</ul>;
}
