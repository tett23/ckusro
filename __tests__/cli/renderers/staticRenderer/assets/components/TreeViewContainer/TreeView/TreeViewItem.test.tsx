import { shallow } from 'enzyme';
import React from 'react';
import TreeViewItem from '../../../../../../../../src/cli/renderers/staticRenderer/assets/components/TreeViewContainer/TreeView/TreeViewItem';
import { buildFileBuffer } from '../../../../../../../__fixtures__';

describe(TreeViewItem, () => {
  it('renders correctly', () => {
    const fileBuffer = buildFileBuffer({ id: 'test_1' });
    const wrapper = shallow(<TreeViewItem fileBuffer={fileBuffer} />);

    expect(wrapper.find('Children').type === null);
    expect(wrapper).toMatchSnapshot();
  });
});
