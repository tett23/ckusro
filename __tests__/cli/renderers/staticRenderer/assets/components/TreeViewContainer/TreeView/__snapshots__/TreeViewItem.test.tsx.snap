// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeViewItem renders correctly 1`] = `ShallowWrapper {}`;
